.github/workflows/reusable-api-e2e.yml

```diff
@@ -15,6 +15,11 @@ on:
         required: false
         default: false
         type: boolean
+      submodule_branch:
+        description: 'Submodule branch to checkout to'
+        required: false
+        default: 'main'
```

Comment:
* I could not create `submodule_branch` as required because `submodules` is not required.
* i added submodule_branch default as a guard but it should not be used IMO.